solidity/contracts/converter/types/liquidity-pool-v3/LiquidityPoolV3Converter.sol

```diff
@@ -0,0 +1,1356 @@
+    */
+    function acceptAnchorOwnership() public virtual override ownerOnly {
+        // verify the the converter has exactly two reserves
+        require(reserveTokenCount() == 2, "ERR_INVALID_RESERVE_COUNT");
+        anchor.acceptOwnership();
+        syncReserveBalances();
+        emit Activation(converterType(), anchor, true);
+    }
+
+    /**
+      * @dev updates the current conversion fee
+      * can only be called by the contract owner
+      *
+      * @param _conversionFee new conversion fee, represented in ppm
+    */
+    function setConversionFee(uint32 _conversionFee) public override ownerOnly {
+        require(_conversionFee <= maxConversionFee, "ERR_INVALID_CONVERSION_FEE");
+        emit ConversionFeeUpdate(conversionFee, _conversionFee);
+        conversionFee = _conversionFee;
+    }
+
+    /**
+      * @dev withdraws tokens held by the converter and sends them to an account
+      * can only be called by the owner
+      * note that reserve tokens can only be withdrawn by the owner while the converter is inactive
+      * unless the owner is the converter upgrader contract
+      *
+      * @param _token   ERC20 token contract address
+      * @param _to      account to receive the new amount
+      * @param _amount  amount to withdraw
+    */
+    function withdrawTokens(IERC20Token _token, address _to, uint256 _amount)
+        public
+        override(IConverter, TokenHolder)
+        protected
+        ownerOnly
+    {
+        address converterUpgrader = addressOf(CONVERTER_UPGRADER);
+        uint256 reserveId = reserveIds[_token];
+
+        // if the token is not a reserve token, allow withdrawal
+        // otherwise verify that the converter is inactive or that the owner is the upgrader contract
+        require(reserveId == 0 || !isActive() || owner == converterUpgrader, "ERR_ACCESS_DENIED");
+        super.withdrawTokens(_token, _to, _amount);
+
+        // if the token is a reserve token, sync the reserve balance
+        if (reserveId != 0)
+            syncReserveBalance(_token);
+    }
+
+    /**
+      * @dev upgrades the converter to the latest version
+      * can only be called by the owner
+      * note that the owner needs to call acceptOwnership on the new converter after the upgrade
+    */
+    function upgrade() public ownerOnly {
+        IConverterUpgrader converterUpgrader = IConverterUpgrader(addressOf(CONVERTER_UPGRADER));
+
+        // trigger de-activation event
+        emit Activation(converterType(), anchor, false);
+
+        transferOwnership(address(converterUpgrader));
+        converterUpgrader.upgrade(version);
+        acceptOwnership();
+    }
+
+    /**
+      * @dev returns the number of reserve tokens defined
+      * note that prior to version 17, you should use 'connectorTokenCount' instead
+      *
+      * @return number of reserve tokens
+    */
+    function reserveTokenCount() public view returns (uint16) {
```

Comment:
maybe it's time to add another function - `reserveTokens` that returns an array of both token addresses.
don't think we use the current `reserveTokens` function anywhere (but need to confirm)

Reply:
Done.
Function name is `reserveTokenArray`, since we already have a function named `reserveTokens` (which returns a reserve-token by index).

Reply:
Fixed (renamed to `reserveTokens `).